dffml/df/memory.py

```diff
@@ -1026,6 +1026,20 @@ def __init__(
         task.add_done_callback(ignore_args(self.completed_event.set))
         return task
 
+    async def dispatch_auto_starts(self, octx: BaseOrchestratorContext, ctx):
+        """
+        """
+        empty_parameter_set = MemoryParameterSet(
+            MemoryParameterSetConfig(ctx=ctx, parameters=[])
+        )
+
+        for operation in octx.config.dataflow.auto_starts.values():
```

Comment:
I don't think we need the `auto_start` parameter. If an operation doesn't have inputs, then the only way it would get run is if we auto start it. Therefore there's really no need to have a separate property to signify that it needs to be auto started, the lack of inputs will signal that to us.

Also, I don't think this is worth creating a separate data structure for. We only care about this when we first start running a dataflow. So we should probably just skip any operations with inputs when we call this function and dispatch all the ones without.

```suggestion
        for operation in octx.config.dataflow.operations.values():
            if operation.inputs:
                continue
```